files/nvidia-overclock.sh

```diff
@@ -6,25 +6,85 @@
 #
 # Overclocking with nvidia-settings
 #
+# Changelog
+# 2018/03/10 Andrea Lanfranchi - https://github.com/AndreaLanfranchi/ethereum_nvidia_miner
+# - Implemented optional per GPU fine tuning
+# - Some cosmetic adjustments for nice output
+#
+
 
 # Load global settings settings.conf
 # shellcheck source=settings.conf
 # shellcheck disable=SC1091
+# shellcheck disable=SC2086
+# shellcheck disable=SC2154
+
 if ! source ~/settings.conf; then
-	echo "FAILURE: Can not load global settings 'settings.conf'"
-	exit 9
+    printf "FAILURE: Can not load global settings 'settings.conf'\n\n"
+    exit 9
 fi
 
 export DISPLAY=:0
 
+# Check if we can set power-limit globally (i.e. with only one call to nvidia-smi) 
+# or we have to invoke it per GPU
+unset MY_WATT_X
+if set -o posix; set | grep -q -E "^MY\_WATT\_[0-9]{1,2}" ; then MY_WATT_X="1"; fi;
+if [ -z ${MY_WATT_X+x} ]; 
+then
+    MY_VAR="MY_WATT"
+    unset MY_VAL
+    if [ ! -z ${!MY_VAR} ] ; then MY_VAL=${!MY_VAR}; fi;
+    if [ ! -z ${MY_VAL+x} ] ; 
+    then
+        printf "\nApplying Power Limit for ALL GPUs \n--------------------------------------------------------------------------------\n" 
+        sudo nvidia-smi -pl "$MY_VAL"
+    fi;
+fi;
+
 # For each graphics card
 nvidia-smi --format=csv,noheader --query-gpu=index | while read -r MY_DEVICE; do
-	# Set Fan speed , Graphics clock, Memory Clock
-	nvidia-settings -a "[gpu:$MY_DEVICE]/GPUPowerMizerMode=1" -a "[gpu:$MY_DEVICE]/GPUFanControlState=1" -a "[fan:$MY_DEVICE]/GPUTargetFanSpeed=$MY_FAN" -a "[gpu:$MY_DEVICE]/GPUGraphicsClockOffset[3]=$MY_CLOCK" -a "[gpu:$MY_DEVICE]/GPUMemoryTransferRateOffset[3]=$MY_MEM"
-	# Set watt/powerlimit. This is also set in miner.sh at autostart.
-	sudo nvidia-smi -i "$MY_DEVICE" -pl "$MY_WATT"
+
+    printf "\n  Processing GPU %s \n  ------------------------------------------------------------------------------\n" "$MY_DEVICE"
+
+    # Init arguments
+    # PowerMizer always se
+    MY_ARG=" -a \"[gpu:$MY_DEVICE]/GPUPowerMizerMode=1\""
```

Comment:
MY_ARG=" -a \"[gpu:$MY_DEVICE]/GPUPowerMizerMode=1\" -a "[gpu:$MY_DEVICE]/GPUFanControlState=1" ".

Without GPUFanControlState Asus GTX1070 do not accept fan value.

Reply:
Added: https://github.com/Cyclenerd/ethereum_nvidia_miner/commit/34117053db537abd0445513a1f9c040cf6a8ed5d

Reply:
Changed: https://github.com/Cyclenerd/ethereum_nvidia_miner/commit/0412c90edf90981e4c4927eaa7b927df3ff50873

Reply:
And: https://github.com/Cyclenerd/ethereum_nvidia_miner/commit/57d2a1601b43df565937c879c36c850e66e6bb9e